src/react/hooks/paragon/useParagonThemeCore.js

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+
+import { logError, logInfo } from '../../../logging';
+import { fallbackThemeUrl, removeExistingLinks } from './utils';
+
+/**
+ * Adds/updates a `<link>` element in the HTML document to load the core application theme CSS.
+ *
+ * @memberof module:React
+ *
+ * @param {object} args
+ * @param {object} args.themeCore Object representing the core Paragon theme CSS.
+ * @param {string} args.onLoad A callback function called when the core theme CSS is loaded.
+ */
+const useParagonThemeCore = ({
+  themeCore,
+  onLoad,
+}) => {
+  const [isParagonThemeCoreLoaded, setIsParagonThemeCoreLoaded] = useState(false);
+  const [isBrandThemeCoreLoaded, setIsBrandThemeCoreLoaded] = useState(false);
+
+  useEffect(() => {
+    // Call `onLoad` once both the paragon and brand theme core are loaded.
+    if (isParagonThemeCoreLoaded && isBrandThemeCoreLoaded) {
+      onLoad();
+    }
+  }, [isParagonThemeCoreLoaded, isBrandThemeCoreLoaded, onLoad]);
+
+  useEffect(() => {
+    // If there is no config for the core theme url, do nothing.
+    if (!themeCore?.urls) {
+      setIsParagonThemeCoreLoaded(true);
+      setIsBrandThemeCoreLoaded(true);
+      return;
+    }
+    const getParagonThemeCoreLink = () => document.head.querySelector('link[data-paragon-theme-core="true"]');
+    const existingCoreThemeLink = document.head.querySelector(`link[href='${themeCore.urls.default}']`);
+    const brandCoreLink = document.head.querySelector(`link[href='${themeCore.urls.brandOverride}']`);
+    if (!existingCoreThemeLink) {
+      const getExistingCoreThemeLinks = (isBrandOverride) => {
+        const coreThemeLinkSelector = `link[data-${isBrandOverride ? 'brand' : 'paragon'}-theme-core="true"]`;
+        return document.head.querySelectorAll(coreThemeLinkSelector);
+      };
+      const createCoreThemeLink = (
+        url,
+        {
+          isFallbackThemeUrl = false,
+          isBrandOverride = false,
+        } = {},
+      ) => {
+        let coreThemeLink = document.createElement('link');
+        coreThemeLink.href = url;
+        coreThemeLink.rel = 'stylesheet';
+        if (isBrandOverride) {
+          coreThemeLink.dataset.brandThemeCore = true;
+        } else {
+          coreThemeLink.dataset.paragonThemeCore = true;
+        }
+        coreThemeLink.onload = () => {
+          if (isBrandOverride) {
+            setIsBrandThemeCoreLoaded(true);
+          } else {
+            setIsParagonThemeCoreLoaded(true);
+          }
+        };
+        coreThemeLink.onerror = () => {
+          logError(`Failed to load core theme CSS from ${url}`);
+          if (isFallbackThemeUrl) {
+            logError(`Could not load core theme CSS from ${url} or fallback URL. Aborting.`);
```

Comment:
[curious] Any concerns around having two `logError` function calls within the same code path? I.e., if `isFallbackThemeUrl` is truthy here, would we have logged basically the same error twice (with slightly differing error messages)?

Reply:
Should we just left the `isFallbackThemeUrl` and relay on the fallback logInfo as the can not load the default URL?

https://github.com/openedx/frontend-platform/pull/689/files#diff-6e8530492255f6f52120a1dd9f69147538c94d6e37be13541f802a2189f80af8R83

Reply:
> We can have a first message (in the `logInfo` - we also can replace it with a `logError`) that is 
**_Could not load core theme CSS from ${url}. Falling back to locally installed core theme CSS: ${coreThemeFallbackUrl}_**
> 
> and then with the `isFallbackThemeUrl` the second message
**_Could not load core theme fallback URL. Aborting._**
> 
> What do you think?

This sounds/looks perfect to me. Thank you so much for the update!